src/firefly/js/ui/SimpleComponent.jsx

```diff
@@ -254,3 +256,32 @@ export function useDebugCycle({id, render=true, mount=true}) {
         render && console.log(id, 'rendering');
     });
 }
+
+/*
+ Enforces our usage of slotProps as a way to define both the slot's component as well as its props.
+ The default props and passed in props are deep merged using smartMerge.
+ <Need to revisit> smartMerge works for now, but ultimately, default props should use a path-based
+    key, so it can be precisely inject the default value without uncertainty.  e.g. webutil's updateSet or setIf based on lodash's set.
+
+ When slotProps is used, it will interpret as:
+ slotProps: {
+   component: elementType,  // when defined, it is used to render the props.  if null, nothing will render in this slot.
+   ...otherProps: any       // these props will be passed to the rendering component
+ }
+
+ Example:
+ const defProps= {text: 'hello world', customMsg: 'none'};
+ const passedInSlotProps = {
+    component: MyResults,
+    customMsg: 'I am Joe'
+ }
+
+ <Slot component={EmptyResults} {...defProps} slotProps={passedInSlotProps}/>
+ This will return:
+   <MyResults text='hello world' customMsg='I am Joe'/>
+ */
+export function Slot({component, slotProps={}, ...props}) {
+    const {component:Component=component, ...pProps} = slotProps;
+    props = smartMerge(props, pProps);
```

Comment:
This will ensure that deep merging of sx I was doing in LandingPage, right? (visually, things are looking same)

Reply:
`smartMerge` will merge plain object and array.  But, having that logic here allow it to change without having to change it everywhere.